### PR TITLE
correct assert error

### PR DESCRIPTION
**Pull request by mylxiaoyi**
call split_load_kwargs with inputs to correct assert error and update parameters of self.module

### CODE REVIEW

File: gluoncv/utils/parallel.py [hunk 1]
```diff
@@ -93,12 +93,13 @@ def __init__(self, module, ctx_list=None, sync=False):
     def __call__(self, inputs, *targets, **kwargs):
         # the inputs should be the outputs of DataParallelModel
         if not self.ctx_list:
-            return self.module(inputs, *targets, **kwargs)
+            return self.module(*inputs, *targets, **kwargs)
```
Comment (zhanghang1989) [inline gluoncv/utils/parallel.py]: *(inputs + targets)

File: gluoncv/utils/parallel.py [hunk 2]
```diff
@@ -93,12 +93,13 @@ def __init__(self, module, ctx_list=None, sync=False):
...
+        inputs, _ = split_load_kwargs(inputs, kwargs, self.ctx_list)
```
Comment (zhanghang1989) [inline gluoncv/utils/parallel.py]: The inputs should be the output of DataParallelModel, which is already split loaded. What's the motivation for this change?

File: gluoncv/utils/parallel.py (hunk 1 above)
Comment (mylxiaoyi) [reply gluoncv/utils/parallel.py]: But when use aux=True in FCN, the input is a tuple. So how to solve it?

Comment (zhanghang1989) [reply gluoncv/utils/parallel.py]: Thanks! I see the problem. A simple solution is
```
if isinstance(input, NDArray):
    return self.module(inputs, *targets, **kwargs)
else:
    return self.module(*(inputs + targets), **kwargs)
```

File: gluoncv/utils/parallel.py (hunk 2 above)
Comment (mylxiaoyi) [reply gluoncv/utils/parallel.py]: I have only one GPU,  when used with aux=True in FCN,  the outputs is a tuple, and the length of the outputs is 2, so cause the assert error, because the len(self.ctx_list) is 1. If used with aux=False in FCN, the length of the outputs is the batch_size, e.g. 4, and also cause assert error.

Comment (zhanghang1989) [reply gluoncv/utils/parallel.py]: Get you. I will take into this. Thx

**Comment by zhanghang1989** [comment]
Addressed in PR https://github.com/dmlc/gluon-cv/pull/98
